mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/ui/game/ChessBoardRobot.kt

```diff
@@ -0,0 +1,194 @@
+package ch.epfl.sdp.mobile.test.ui.game
+
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.ComposeTestRule
+import ch.epfl.sdp.mobile.test.ui.AbstractRobot
+import ch.epfl.sdp.mobile.ui.game.ChessBoardState
+import ch.epfl.sdp.mobile.ui.i18n.LocalizedStrings
+
+/**
+ * An interface representing the operations which are available when interacting with the chess
+ * board. Inputs are performed in a coordinate system starting at the top left of the chessboard,
+ * with the x axis going towards the right and the y axis going towards the bottom.
+ *
+ * If the x or y coordinates are outside of the [0..8) range, inputs will NOT be clamped. This means
+ * you can drop pieces outside of the board.
+ */
+interface ChessBoardRobotInputScope {
+
+  /**
+   * Puts the pointer with the given id down on this cell.
+   *
+   * @param x the x coordinate at which the pointer goes down.
+   * @param y the y coordinate at which the pointer goes down.
+   * @param pointerId the identifier of the pointer.
+   */
+  fun down(x: Int, y: Int, pointerId: Int = 0)
+
+  /**
+   * Moves the pointer with the given id to this cell.
+   *
+   * @param x the x coordinate of the target cell.
+   * @param y the y coordinate of the target cell.
+   * @param pointerId the identifier of the pointer.
+   */
+  fun moveTo(x: Int, y: Int, pointerId: Int = 0)
+
+  /**
+   * Moves the pointer with the given id by a relative amount of cells.
```

Comment:
Question: what do you mean by pointer?

Reply:
On multi-touch devices, gestures are composed of multiple pointers (usually the fingers of the user). This terminology is used by Android gesture system (see the documentation for [`MotionEvent`](https://developer.android.com/reference/android/view/MotionEvent) for instance). Compose uses the same terminology.

The default pointer has value 0. If you look into Jetpack Compose's `SemanticsNodeInteraction.performTouchInput { }` method, you'll see that gestures are performed with this pointer by default, but allow the user to specify another pointer identifier. The `ChessBoardRobot` essentially allows the same behaviour, so we can test multi-touch gestures too.